cel/options.go

```diff
@@ -376,3 +378,83 @@ func EvalOptions(opts ...EvalOption) ProgramOption {
 		return p, nil
 	}
 }
+
+func fieldToCELType(field protoreflect.FieldDescriptor) (*exprpb.Type, error) {
+	// https://github.com/google/cel-spec/blob/master/doc/langdef.md#protocol-buffer-data-conversion
+	switch (field).Kind() {
+	case protoreflect.Int32Kind, protoreflect.Int64Kind, protoreflect.Sint32Kind, protoreflect.Sint64Kind, protoreflect.Sfixed32Kind, protoreflect.Sfixed64Kind, protoreflect.EnumKind:
+		return decls.Int, nil
+	case protoreflect.Uint32Kind, protoreflect.Uint64Kind, protoreflect.Fixed32Kind, protoreflect.Fixed64Kind:
+		return decls.Uint, nil
+	case protoreflect.FloatKind, protoreflect.DoubleKind:
+		return decls.Double, nil
+	case protoreflect.BoolKind:
+		return decls.Bool, nil
+	case protoreflect.StringKind:
+		return decls.String, nil
+	case protoreflect.BytesKind:
+		return decls.Bytes, nil
+	case protoreflect.MessageKind:
+		// Special case well-known types
+		// https://github.com/googleapis/googleapis/blob/b6b4b89cae9b4734b7810c65a4fa69b5e9083cdc/google/api/expr/v1alpha1/checked.proto#L132
+		if (field).Message().FullName() == "google.protobuf.Any" {
+			return decls.Any, nil
+		} else if (field).Message().FullName() == "google.protobuf.Timestamp" {
+			return decls.Timestamp, nil
+		} else if (field).Message().FullName() == "google.protobuf.Duration" {
+			return decls.Duration, nil
+		} else {
+			return decls.NewObjectType(string((field).Message().FullName())), nil
+		}
+	default:
+		return nil, fmt.Errorf("field %s type %s not implemented", (field).FullName(), (field).Kind().String())
+	}
+}
+
+func fieldToDecl(field protoreflect.FieldDescriptor) (*exprpb.Decl, error) {
+	name := string(field.Name())
+	if (field).IsMap() {
+		mapKey := (field).MapKey()
+		mapValue := (field).MapValue()
+		keyType, err := fieldToCELType(mapKey)
+		if err != nil {
+			return nil, err
+		}
+		valueType, err := fieldToCELType(mapValue)
+		if err != nil {
+			return nil, err
+		}
+		return decls.NewVar(name, decls.NewMapType(keyType, valueType)), nil
+	} else if (field).IsList() {
+		elemType, err := fieldToCELType(field)
+		if err != nil {
+			return nil, err
+		}
+		return decls.NewVar(name, decls.NewListType(elemType)), nil
+	} else {
+		celType, err := fieldToCELType(field)
+		if err != nil {
+			return nil, err
+		}
+		return decls.NewVar(name, celType), nil
+	}
+}
+
+// DeclareContextProto returns an option to extend CEL environment with declarations from the given context proto (https://github.com/google/cel-spec/blob/master/doc/langdef.md#evaluation-environment). Each field of the proto defines a variable of the same name in the environment.
```

Comment:
Can you please break the comment up? The first line should probably end at 'given context proto' with the link being on the last line as a further reference.